docs/content/features/auto-responses/index.md

```diff
@@ -180,6 +188,108 @@ In the event that architus is unable to parse your trigger regex, it will give a
+
+As an example:
+```py
+print([a for a in "aoua".elems()])
+```
+will print `["a", "o", "u", "a"]`.
+
+### Architus Specifics
+To improve user experience, some changes to the Starlark language have been made. Specifically:
+- Python style `while` loops have been added
+- Top level `if` statements are allowed
+- Global variables are mutable
+- Recursive function calls are allowed
+- Function declarations can be nested
+
+### Builtin Functions
+The architus implementation of Starlark adds in a few extra builtins to improve user experience. They are called the same as any other Python builtin with the same behavior.
+
+#### Random
+The `random` function can be called with no arguments to return a random float in the range \[0,1\).
+
+For example:
+```py
+print(random())
+```
+may print `0.686645146372598`.
+
+#### Randint
+The `randint` function can be called with a high and low parameter to return a random integer in that range. For instance, `randint(lo, hi)` will return an integer in the range \[lo, hi\).
+
+#### Choice
+The `choice` function will return a random element from a list. It accepts a single list element as an argument and its return type can be any of the types in the list.
+
+For example:
+```py
+print(choice(["hello", 'a', 1, 1.2345, randint]))
+```
+may print `a`.
+
+#### Sine
+The `sin` function is the standard trigonometric sine function. It accepts radians as an argument and will return a float in the range \[0, 1\].
+
+#### Struct
+The `struct` function will produce a class like data structure that can be used to store data. Unlike Python, Starlark does not support classes and objects. To make up for this, our implementation of Starlark adds in a struct datatype that can be used in a similar manner.
+
+A struct is purely a data structure and does not support any methods that can act on the structure. The function will take some number of keyword arguments and return a struct where each keyword is a member of the struct and the argument passed to that keyword is the value associated with that member.
+
+Example:
+```py
+book = struct(author="Frank Herbert", title="Dune", pages=818, reviews=["great", "amazing"])
+assert(book.author=="Frank Herbert")
+assert(book.pages==818)
+```
+
+#### Sum
+The `sum` function will sum all of the elements of an iterable. It accepts a single parameter that is the iterable value to sum over. `sum` requires that the parameter passed in is solely composed of numbers: integers, floats, or both. If only integers are in the iterable, it will return an integer. However, if the iterable is composed of floats or integers and floats, it will return a float value. The sum will always start at 0 and sum from there.
```

Comment:
this is too detailed.  no one who knows what an iterator is doesn't already know what sum does, and it's unlikely anyone cares about the type output